homeassistant/components/samsungtv/config_flow.py

```diff
@@ -99,81 +153,85 @@ async def async_step_import(self, user_input=None):
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         if user_input is not None:
-            ip_address = await self.hass.async_add_executor_job(
-                _get_ip, user_input[CONF_HOST]
-            )
-
-            await self.async_set_unique_id(ip_address)
-            self._abort_if_unique_id_configured()
-
-            self._host = user_input.get(CONF_HOST)
-            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
-            self._name = user_input.get(CONF_NAME)
+            self._host = user_input[CONF_HOST]
+            self._name = user_input[CONF_NAME]
             self._title = self._name
 
-            result = await self.hass.async_add_executor_job(self._try_connect)
+            await self._abort_if_already_configured()
+
+            await self.hass.async_add_executor_job(self._try_connect)
+            await self._get_and_check_device_info()
```

Comment:
Could you set the unique ID after you get the device info?

Reply:
Unique id call added